[Bug]: Storybook hangs/doesn't finish building 8.4.6/8.4.7

### Describe the bug

We're using Storybook (with React) in version 8.4.5. I noticed that our Renovate update failed for 8.4.6 and now 8.4.7 as well. I didn't create a bug ticket so far, because I cannot reproduce it locally and only see it in our CI (Gitlab).

I decided to create a ticket now as I was checking if someone else has the same problem, but I couldn't find something related. Maybe it helps someone else to jump into the discussion (and ideally a solution later). (https://github.com/storybookjs/storybook/issues/29715 seems to about a different issue.)

### Reproduction link

#sorry-:-(

### Reproduction steps

_No response_

### System

Storybook in the mentioned versions, Gitlab CI and `$ node -v v22.11.0`.

### Additional context

The last output I get after running `$ storybook build--output-dir storybook` is this:

> Generate File to /builds/xxx/yyy/storybook/meta.json
info => Preview built (1.33 min)
info => Output directory: /builds/xxx/yyy/storybook
attention => Storybook now collects completely anonymous telemetry regarding usage.
This information is used to shape Storybook's roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://storybook.js.org/telemetry

Afterwards my CI job runs into a timeout. As far as

I tried to get more information by using different log levels and --debug, but I couldn't get more information. I will try to debug more and if I find a solution/workaround I will update (and close) this issue.

Hi, I came across this issue because as @donaldpipowitch I get stuck while upgrading to v8.4.7.

The problem on my side is that while starting the development I got a message `error TS5083: Cannot read file './tsconfig.json'.`

```shell
@storybook/core v8.4.7

The CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.
info => Starting manager..
info => Starting preview..
error TS5083: Cannot read file './tsconfig.json'.
```

When adding a `tsconfig.json` file with just `{}` the problem disappears.

However, after reading @yannbf [answer](https://github.com/storybookjs/storybook/issues/29828#issuecomment-2560964989) in this thread I have tried to change `react-docgen-typescript` to just `react-docgen` and it looks like it is working even without the overrides.

Just in case these two are related:
- PR with the upgrade - https://github.com/lmc-eu/spirit-design-system/pull/1849
- Storybook configuration - https://github.com/lmc-eu/spirit-design-system/blob/main/apps/storybook/main.ts

In case this is a separate issue, let me know. Thanks.

--- a/MIGRATION.md
+++ b/MIGRATION.md
@@ -465,7 +465,9 @@ Storybook now supports [Angular's experimental zoneless mode](https://angular.de
 
 ### React Vite: react-docgen-typescript is updated
 
-Storybook now uses [react-docgen-typescript](https://github.com/joshwooding/vite-plugin-react-docgen-typescript) v0.5.0 which updates its internal logic on how it parses files. Depending on how big is your codebase and which version of TypeScript you use, you might have performance issues. If that is the case, you can opt-in to a newer implementation under the `EXPERIMENTAL_useWatchProgram` flag. Keep in mind that this flag is experimental and also does not support the `references` field in tsconfig.json files.
+Storybook now uses [react-docgen-typescript](https://github.com/joshwooding/vite-plugin-react-docgen-typescript) v0.5.0 which updates its internal logic on how it parses files, available under an experimental feature flag `EXPERIMENTAL_useWatchProgram`, which is disabled by default.
+
+Previously, once you made changes to a component's props, the controls and args table would not update unless you restarted Storybook. With the `EXPERIMENTAL_useWatchProgram` flag, you do not need to restart Storybook anymore, however you do need to refresh the browser page. Keep in mind that this flag is experimental and also does not support the `references` field in tsconfig.json files. Depending on how big your codebase is, it might have performance issues.
 
 ```ts
 // .storybook/main.ts

--- a/MIGRATION.md
+++ b/MIGRATION.md
@@ -2,6 +2,9 @@
 
 - [From version 8.5.x to 8.6.x](#from-version-85x-to-86x)
   - [Angular: Support experimental zoneless support](#angular-support-experimental-zoneless-support)
+- [From version 8.4.x to 8.5.x](#from-version-84x-to-85x)
+  - [React Vite: react-docgen-typescript is updated](#react-vite-react-docgen-typescript-is-updated)
+  - [Introducing features.developmentModeForBuild](#introducing-featuresdevelopmentmodeforbuild)
   - [Added source code panel to docs](#added-source-code-panel-to-docs)
   - [Addon-a11y: Component test integration](#addon-a11y-component-test-integration)
   - [Addon-a11y: Changing the default element selector](#addon-a11y-changing-the-default-element-selector)
@@ -460,6 +463,24 @@ Storybook now supports [Angular's experimental zoneless mode](https://angular.de
 
 ## From version 8.4.x to 8.5.x
 
+### React Vite: react-docgen-typescript is updated
+
+Storybook now uses [react-docgen-typescript](https://github.com/joshwooding/vite-plugin-react-docgen-typescript) v0.5.0 which updates its internal logic on how it parses files. Depending on how big is your codebase and which version of TypeScript you use, you might have performance issues. If that is the case, you can opt-in to a newer implementation under the `EXPERIMENTAL_useWatchProgram` flag. Keep in mind that this flag is experimental and also does not support the `references` field in tsconfig.json files.
+
+```ts
+// .storybook/main.ts
+const config = {
+  // ...
+  typescript: {
+    reactDocgen: "react-docgen-typescript",
+    reactDocgenTypescriptOptions: {
+      EXPERIMENTAL_useWatchProgram: true,
+    },
+  },
+};
+export default config;
+```
+
 ### Introducing features.developmentModeForBuild
 
 As part of our ongoing efforts to improve the testability and debuggability of Storybook, we are introducing a new feature flag: `developmentModeForBuild`. This feature flag allows you to set `process.env.NODE_ENV` to `development` in built Storybooks, enabling development-related optimizations that are typically disabled in production builds.
@@ -473,7 +494,7 @@ export default {
     developmentModeForBuild: true,
   },
 };
-````
+```
 
 ### Added source code panel to docs
 

--- a/code/frameworks/react-native-web-vite/package.json
+++ b/code/frameworks/react-native-web-vite/package.json
@@ -56,7 +56,7 @@
     "@babel/plugin-transform-modules-commonjs": "^7.26.3",
     "@babel/preset-react": "^7.26.3",
     "@bunchtogether/vite-plugin-flow": "^1.0.2",
-    "@joshwooding/vite-plugin-react-docgen-typescript": "0.4.2",
+    "@joshwooding/vite-plugin-react-docgen-typescript": "0.5.0",
     "@storybook/builder-vite": "workspace:*",
     "@storybook/react": "workspace:*",
     "@storybook/react-vite": "workspace:*",

--- a/code/frameworks/react-vite/package.json
+++ b/code/frameworks/react-vite/package.json
@@ -57,7 +57,7 @@
     "prep": "jiti ../../../scripts/prepare/bundle.ts"
   },
   "dependencies": {
-    "@joshwooding/vite-plugin-react-docgen-typescript": "0.4.2",
+    "@joshwooding/vite-plugin-react-docgen-typescript": "0.5.0",
     "@rollup/pluginutils": "^5.0.2",
     "@storybook/builder-vite": "workspace:*",
     "@storybook/react": "workspace:*",

--- a/docs/api/main-config/main-config-typescript.mdx
+++ b/docs/api/main-config/main-config-typescript.mdx
@@ -79,7 +79,7 @@ Options to pass to `fork-ts-checker-webpack-plugin`, if [enabled](#check). See [
 
   Type: `ReactDocgenTypescriptOptions`
 
-  Configures the options to pass to `react-docgen-typescript-plugin` if `react-docgen-typescript` is enabled. See [docs for available options](https://github.com/hipstersmoothie/react-docgen-typescript-plugin).
+  Configures the options to pass to `react-docgen-typescript-plugin` if `react-docgen-typescript` is enabled. See docs for available options [for Webpack projects](https://github.com/hipstersmoothie/react-docgen-typescript-plugin) or [for Vite projects](https://github.com/joshwooding/vite-plugin-react-docgen-typescript).
 
   {/* prettier-ignore-start */}
 

